atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -498,27 +500,27 @@ private void removeFromBlacklistAfterResponse(CassandraServer cassandraServer) {
         }
     }
 
+    private void logTokenRingOnStartup() {
+        Multimap<Set<TokenRange>, CassandraServer> tokenRangesToServer = getTokenRangesFromAllReachableServers();
+
+        if (tokenRangesToServer == null) {
+            log.info("Could not load token ring upon start up!");
+            return;
+        }
+
+        log.info(
+                "This is the token Range as seen by init Cassandra server pool\n{}",
+                SafeArg.of("tokenRangesToServer", CassandraLogHelper.tokenRangesToServer(tokenRangesToServer)));
```

Comment:
This is NOT SAFE

Reply:
wait, maybe it is

Reply:
nope, it's not. Anything that logs tokenRanges directly without hashing the endpoints is totally unsafe